packages/web-component/src/lib/descope-wc/BaseDescopeWc.ts

```diff
@@ -155,18 +155,52 @@ class BaseDescopeWc extends HTMLElement {
         string,
         any
       >;
-      return Object.entries(form).reduce(
-        (prev, [key, value]) => ({
+
+      let formData = form;
+      const vals = Object.values(form);
+
+      // transform values to object structure if needed:
+      // we want to support the existing API for simple value replacement, and allow using
+      // a new API for more complex settings (like attribute overrides).
+      // the existing API consists of key/value map (e.g. `'{"email":"my@email.com"}'`)
+      // while the new API is a structured object (e.g. `'{ "email": { "value": "my@email.com", "disabled": "true" } }'`)
+      if (vals.every((s) => typeof s === 'string')) {
+        formData = Object.fromEntries(
+          Object.keys(form).map((key) => [key, { value: form[key] }]),
+        );
+        // enforce a single type of configuration, return an empty object if some values
+        // are in objects and others are string
+      } else if (
+        vals.some((j) => typeof j === 'object') &&
+        vals.some((j) => typeof j === 'string')
+      ) {
+        // eslint-disable-next-line no-console
+        console.error('flow inputs `form` attribute mismatch configuration');
+        return {};
+      }
+
+      return Object.entries(formData).reduce(
+        (prev, [name, value]) => ({
           ...prev,
-          [`form.${key}`]: value,
+          [`form.${name}`]: value,
         }),
-        form,
+        formData,
       );
```

Comment:
i dont think we need to force single type configurations. You can probably add validation making sure its a string or an object with `value` before, but this should always be good:

```
return Object.entries(form).reduce(
        (prev, [name, value]) => ({
          ...prev,
          [`form.${name}`]: typeof value === 'string' ? { value } : {...value},
        }),
        formData,
      );
```

also mind that `typeof j === 'object'` is true for functions and null objects, might cause issues.

Reply:
Not sure I follow - the configuration is backwards compatible, but we transform it here for consistency in all other operations.
the user can provide the existing configuration, or an object structure.

I'll take care of the `typeof` issue, thanks.

Reply:
Also, [`typeof () => {}` returns `Function`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#description), no?

Reply:
@tomerlichtash we can easily allow `{ email: "test@descope.com", email2: { value: "test@descope.com", disabled: true }}` - this is why i suggested just
```
return Object.entries(form).reduce(
        (prev, [name, value]) => ({
          ...prev,
          [`form.${name}`]: typeof value === 'string' ? { value } : {...value},
        }),
        form,
      );
```

Reply:
IMHO seems inconsistent... what are the benefits of allowing this? Why not keep a single structure?

Reply:
there is no real reason to prevent this, even for testing purposes. forcing to change the object for consistency is futile. mind that also if value is empty maybe you need to force it to empty string to avoid undefined values

Reply:
Changed to support both formats, added test as well.